AWSAuthSDK/Tests/AWSMobileClientTests/AWSMobileClientTests.swift

```diff
@@ -136,33 +136,67 @@ class AWSMobileClientTests: XCTestCase {
             return nil
         }).waitUntilFinished()
     }
-    
-    func signIn(username: String) {
+
+    func signIn(username: String, password: String? = nil, verifySignState: SignInState = .signedIn) {
+        let passwordToUse = password ?? sharedPassword
         let signInWasSuccessful = expectation(description: "signIn was successful")
-        AWSMobileClient.sharedInstance().signIn(username: username, password: sharedPassword) { (signInResult, error) in
+        AWSMobileClient.sharedInstance().signIn(username: username, password: passwordToUse) { (signInResult, error) in
             if let error = error {
                 XCTFail("User login failed: \(error.localizedDescription)")
                 return
             }
 
             if let signInResult = signInResult {
-                switch(signInResult.signInState) {
-                case .signedIn:
-                    print("signed in")
-                default:
-                    XCTFail("User was not signed in, un-expected.")
-                }
+                XCTAssertEqual(signInResult.signInState, verifySignState, "Could not verify sign in state")
             }
             signInWasSuccessful.fulfill()
         }
         wait(for: [signInWasSuccessful], timeout: 5)
     }
     
+    func confirmSign(challengeResponse: String, userAttributes:[String:String] = [:], verifySignState: SignInState = .signedIn) {
+        
+        let signInConfirmWasSuccessful = expectation(description: "signIn confirm was successful")
+        AWSMobileClient.sharedInstance().confirmSignIn(challengeResponse: challengeResponse,
+                                                       userAttributes: userAttributes) {
```

Comment:
As noted, we should also add a test asserting the user attributes are passed through

Reply:
Talked offline, this isn't needed.